web-common/src/features/chat/core/messages/Messages.svelte

```diff
@@ -86,108 +85,109 @@
+          duration={block.duration}
+          {tools}
+        />
+      {:else if block.type === "working"}
+        <WorkingBlock />
+      {:else if block.type === "chart"}
+        <ChartBlock
+          message={block.message}
+          resultMessage={block.resultMessage}
+          chartType={block.chartType}
+          chartSpec={block.chartSpec}
+          {tools}
+        />
+      {:else if block.type === "file-diff"}
+        <FileDiffBlock
+          message={block.message}
+          resultMessage={block.resultMessage}
+          filePath={block.filePath}
+          diff={block.diff}
+          isNewFile={block.isNewFile}
+          {tools}
+        />
+      {/if}
     {/each}
   {/if}
-  {#if isStreaming}
-    <div class="response-loading">
-      <LoadingSpinner size="1.2em" />
-      Thinking...
-    </div>
-  {:else if hasStreamError}
+  {#if hasStreamError}
     <Error headline="Failed to generate response" error={$streamErrorStore} />
   {/if}
 </div>
 
 <style lang="postcss">
   .chat-messages {
-    flex: 1;
-    display: flex;
-    flex-direction: column;
-    gap: 0.5rem;
+    @apply flex-1;
+    @apply flex flex-col gap-2;
     background: var(--surface);
   }
 
-  /* Sidebar layout: messages container scrolls */
   .chat-messages.sidebar {
-    overflow-y: auto;
-    padding: 0rem 1rem 0rem 1rem;
+    @apply overflow-y-auto;
+    @apply px-4;
   }
 
-  /* Fullpage layout: parent scrolls, content centered */
   .chat-messages.fullpage {
-    padding: 1rem 1rem;
-    max-width: 48rem;
-    margin: 0 auto;
-    width: 100%;
-    min-height: 100%;
+    @apply p-4;
+    @apply max-w-3xl mx-auto w-full;
+    @apply min-h-full;
   }
 
   .chat-empty {
-    display: flex;
-    flex-direction: column;
-    align-items: center;
-    justify-content: center;
-    height: 100%;
-    text-align: center;
-    color: #6b7280;
+    @apply flex flex-col;
+    @apply items-center justify-center;
+    @apply h-full text-center;
+    @apply text-gray-500;
   }
 
-  /* Fullpage layout: enhanced empty state */
   .chat-messages.fullpage .chat-empty {
-    padding: 4rem 2rem;
+    @apply py-16 px-8;
   }
 
   .chat-empty-title {
-    font-size: 1rem;
-    font-weight: 600;
-    color: #374151;
-    margin-bottom: 0.25rem;
+    @apply text-base font-semibold;
+    @apply text-gray-700 mb-1;
   }
 
   .chat-messages.fullpage .chat-empty-title {
-    font-size: 1.5rem;
-    font-weight: 600;
-    color: #111827;
-    margin-bottom: 0.5rem;
+    @apply text-2xl font-semibold;
+    @apply text-gray-900 mb-2;
   }
 
   .chat-empty-subtitle {
-    font-size: 0.75rem;
-    color: #6b7280;
+    @apply text-xs text-gray-500;
   }
 
   .chat-messages.fullpage .chat-empty-subtitle {
-    font-size: 1rem;
-    color: #6b7280;
+    @apply text-base text-gray-500;
   }
 
-  /* Responsive: adjust fullpage empty state for mobile */
   @media (max-width: 640px) {
     .chat-messages.fullpage .chat-empty {
-      padding: 2rem 1rem;
+      padding-top: 2rem;
```

Comment:
nit: use tailwind styles

Reply:
Unfortunately, within a media query, `@apply <tailwind classes>` leads to this (cryptic) error: 
<img width="1366" height="365" alt="image" src="https://github.com/user-attachments/assets/00323bdb-2db4-49cf-9a36-7268a1225893" />

For now, IMO we should live with this Svelte quirk and be okay with raw CSS within media queries.